### PR TITLE
soc: stm32: Enable cortex-m systick timer by default

### PR DESCRIPTION
**Pull request by erwango**
Move systick activation in soc/ as a Kconfig.defconfig file and
remove activation in boards _defconfig files.
This will allow to deactivate it in a more flexible way
with upcoming LPTIMER as tick source when power management
features are enabled.

### CODE REVIEW

**Review by avisconti — APPROVED** [review]
LGTM
So, is this valid for really all stm32 SoCs?

**Comment by erwango** [comment]
> So, is this valid for really all stm32 SoCs?

since there is  no alterntate tick source available today, it has to be.

**Comment by erwango** [comment]
@FRASTM

**Review by MaureenHelm — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Comment by erwango** [comment]
@MaureenHelm , can we get this one merged, as we'll never manage to get shippable passed.
FYI, I've run a ` ./scripts/sanitycheck -T samples/hello_world/ -l -N ;` showing no impact.

**Review by ioannisg — COMMENTED** [review]
@erwango I am not sure we actually need a comment to make it clear that Cortex-M0 does not have SysTick by default.
If so, then do we need also a comment that VTOR, SPLIM registers, etc. are not available by default?

Comments don't hurt, but we should be consistent on how we describe HW capabilities. Not selecting the options sufficiently indicates that the core does not support it.

You could add a help text where the CPU_CORTEX_M_HAS_SYSTICK is defined, to stress that the HW is optional

File: soc/arm/st_stm32/common/Kconfig.defconfig.series [hunk 1]
```diff
@@ -9,6 +9,10 @@
 
 if SOC_FAMILY_STM32
 
+# SYSTICK is the default tick source on STM32 Socs
+config CORTEX_M_SYSTICK
+	default y
```
Comment (ioannisg) [inline soc/arm/st_stm32/common/Kconfig.defconfig.series]: SoCs

**Review by ioannisg — COMMENTED** [review]
Regarding the second patch, i.e. moving the CONFIG_CORTEX_M_SYSTIC=y to SoC Kconfig: this is not a problem, in general. However, do you think the option should be selected together with CONFIG_SYS_CLOCK_HW_CYCLES_PER_SECOND? Right now, the latter is still at board level, but this clearly depends on the SysTick, AFAIU

**Comment by erwango** [comment]
@ioannisg , thanks for the review

> You could add a help text where the CPU_CORTEX_M_HAS_SYSTICK is defined, to stress that the HW is optional

Ok, I'll replace the comment by yours.

> do you think the option should be selected together with CONFIG_SYS_CLOCK_HW_CYCLES_PER_SECOND? Right now, the latter is still at board level, but this clearly depends on the SysTick, AFAIU

If the question is "should we move `CONFIG_SYS_CLOCK_HW_CYCLES_PER_SECOND` at SoC level", then the response is:
No, since `CONFIG_SYS_CLOCK_HW_CYCLES_PER_SECOND` depends on each SoC capabilities, but also board configuration itself (either CPU is driven by fixed clock or PLL, and if drivern by PLL, on PLL configuration).

**Pull request merged**